test/cql-pytest/test_permissions.py

```diff
@@ -553,3 +554,174 @@ def test_native_functions_always_exeutable(cql):
                     assert list(user_session.execute(f"SELECT max(a) FROM {table}")) == [(84,)]
                     assert list(user_session.execute(f"SELECT min(a) FROM {table}")) == [(3,)]
                     assert list(user_session.execute(f"SELECT sum(a) FROM {table}")) == [(102,)]
+
+# When a role has permissions to create a table, the role is given full
+# access to the table it just created (we call this feature "auto-grant").
+# The following tests check the auto-grant feature. The first test checks
+# that the base table created by CREATE TABLE is usable by the role creating
+# it. The second test is about the view created by CREATE MATERIALIZED VIEW
+# being accessible. The third test is about newly created CDC being accessible.
+def test_auto_grant_base(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Check that the username was auto-granted SELECT permissions
+                user_session.execute(f'SELECT * FROM {table}')
+                # Check that the username was auto-granted MODIFY permissions
+                user_session.execute(f'INSERT INTO {table}(p) VALUES(42)')
+                # Check that the username was auto-granted ALTER permissions
+                user_session.execute(f"ALTER TABLE {table} WITH comment = 'hey'")
+                # When this scope ends, the test table is dropped by
+                # user_session, so we check that DROP permissions were auto-
+                # granted as well.
+
+# Test that a role creating a materialized view is automatically granted
+# permissions to read it.
+def test_auto_grant_view(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key, v int"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Because username has CREATE permissions, it can also use
+                # CREATE MATERIALIZED VIEW
+                mv = f'{test_keyspace}.{unique_name()}'
+                user_session.execute(f'CREATE MATERIALIZED VIEW {mv} AS SELECT * FROM {table} WHERE p IS NOT NULL AND v IS NOT NULL PRIMARY KEY (v, p)')
```

Comment:
The way I understand the testcode is that the same user creates a table and MV and is auto-granted permissions to modify both.
What if 1st user creates a table, 2nd user gets permissions to this table and 2nd user creates MV - will 1st user also have access to the 2nd user's MV? I suspect yes, because, under the hood, both users would be assigned the same role, correct?

Reply:
The two users in your example, if they are different "users" (roles), will not have the same role, no.

Actually, to create a table on a base table you need ALTER permissions on that table. I suspect (but didn't test, I guess I could) that if user1 creates basetable and gives user2 ALTER permissions on it, and then user2 create view on that basetable, it is user2, not user1, who gets full permissions on the **view**. If this suspicion is true it means among other things that user1 can never delete its base table, because it needs to delete the view first and it can't :-)

I can write a test for this, although to be honest, I don't think anybody even cares what happens in this bizarre case ;-)

Reply:
I agree we don't have to test every corner case at the moment.